diff --git a/csw.py b/csw.py
index ce2f4bc..d8f9d19 100644
--- a/csw.py
+++ b/csw.py
@@ -265,7 +265,8 @@ class CatalogueServiceWeb:
         }
 
         self.request = '%s%s%s' % (bind_url(self.url), urlencode(data), ','.join(id))
-
+        if self.url == "http://find.ga.gov.au/FIND/geonetwork/srv/eng/csw?version=2.0.2&service=CSW":
+            self.request = 'http://find.ga.gov.au/FIND/metadata-record/get-xml?' + urlencode({'uuid':id[0]})
         self._invoke()
  
         if self.exceptionreport is None:
@@ -505,6 +506,10 @@ class CatalogueServiceWeb:
                 val = i.find(util.nspath_eval('gmd:fileIdentifier/gco:CharacterString', namespaces))
                 identifier = self._setidentifierkey(util.testXMLValue(val))
                 self.records[identifier] = MD_Metadata(i)
+            for i in self._exml.findall('.//'+util.nspath_eval('mcp:MD_Metadata', namespaces)):
+                val = i.find(util.nspath_eval('gmd:fileIdentifier/gco:CharacterString', namespaces))
+                identifier = self._setidentifierkey(util.testXMLValue(val))
+                self.records[identifier] = MD_Metadata(i)
         elif outputschema == namespaces['fgdc']: # fgdc csdgm
             for i in self._exml.findall('.//metadata'):
                 val = i.find('idinfo/datasetid')
@@ -580,7 +585,8 @@ class CatalogueServiceWeb:
             self.request = util.xml2string(etree.tostring(self.request))
 
             self.response = util.http_post(self.url, self.request, self.lang, self.timeout)
-
+        if 'get-xml' in self.request:
+            self.response = '<?xml version="1.0" encoding="UTF-8"?><csw:GetRecordByIdResponse xmlns:csw="http://www.opengis.net/cat/csw/2.0.2">' + self.response.replace('<?xml version="1.0" encoding="UTF-8"?>','') + '</csw:GetRecordByIdResponse>'
         # parse result see if it's XML
         self._exml = etree.parse(StringIO.StringIO(self.response))
 
diff --git a/namespaces.py b/namespaces.py
index 6037cdf..c5d66bc 100644
--- a/namespaces.py
+++ b/namespaces.py
@@ -21,6 +21,7 @@ class Namespaces(object):
         'gmx'   :   'http://www.isotc211.org/2005/gmx',
         'gts'   :   'http://www.isotc211.org/2005/gts',
         'ogc'   :   'http://www.opengis.net/ogc',
+        'mcp'   :   'http://bluenet3.antcrc.utas.edu.au/mcp',
         'om'    :   'http://www.opengis.net/om/1.0',
         'om10'  :   'http://www.opengis.net/om/1.0',
         'om100' :   'http://www.opengis.net/om/1.0',
@@ -133,4 +134,4 @@ class Namespaces(object):
             if key in self.namespace_dict.keys():
                 retval[key] = self.namespace_dict[key]
 
-        return retval
\ No newline at end of file
+        return retval
